cobbler/actions/buildiso/standalone.py

```diff
@@ -139,12 +151,18 @@ def _generate_item(
         }
         if descendant_obj.COLLECTION_TYPE == "profile":
             config_args.update({"menu_indent": 0})
-            autoinstall_args = {"profile": descendant_obj}
+            profile_obj = cast("Profile", descendant_obj)
+            isolinux, grub, to_copy = self._generate_descendant_config(**config_args)
+            autoinstall = self.api.autoinstallgen.generate_autoinstall(
+                profile=profile_obj
+            )
         else:  # system
             config_args.update({"menu_indent": 4})
-            autoinstall_args = {"system": descendant_obj}
-        isolinux, grub, to_copy = self._generate_descendant_config(**config_args)
-        autoinstall = self.api.autoinstallgen.generate_autoinstall(**autoinstall_args)
+            system_obj = cast("System", descendant_obj)
+            isolinux, grub, to_copy = self._generate_descendant_config(**config_args)
+            autoinstall = self.api.autoinstallgen.generate_autoinstall(
+                system=system_obj
+            )
```

Comment:
Ah yeah. This is a change I had to do to make `pyright` happy.

The `pyright` checks were raising errors due "descendant_obj" being `Union["Profile", "System"]` and not pyright not able to determine properly when building the arguments to pass to `self.api.autoinstallgen.generate_autoinstall`.

Reply:
I wasn't aware so far of `cast()`. I think this is a rather elegant way to prevent the usage of `# type: ignore`. I like it. Thanks a lot for that. I'll resolve this too to enable merging.

Reply:
I disagree, in my opinion this is not elegant. It's repetitive, the fact that `descendant_obj` is a `Profile` in the `if`-branch is evident from the condition. That, together with the type annotation that `descendant_obj` is either a `Profile` or a `System` makes it clear that the `else`-branch is for a `System`.

All this change does is add noise that everyone reading needs to decipher first (what is `cast()` doing?) while it does not improve anything. I'd prefer `# type: ignore` until `pyright` gets smart enough to see that the types have already been checked. If `s/descendant_obj.COLLECTION_TYPE == "profile"/isinstance(descendant_obj, "Profile")/` helps, that would be a change worth doing to help the type checker.

Reply:
Thanks for your inputs!

I've tried this way and it still doesn't work with `pyright`:

```python
        if isinstance(descendant_obj, "Profile"):
            config_args.update({"menu_indent": 0})
            autoinstall_args = {"profile": descendant_obj}
        else:  # system
            config_args.update({"menu_indent": 4})
            autoinstall_args = {"system": descendant_obj}
        isolinux, grub, to_copy = self._generate_descendant_config(**config_args)
        autoinstall = self.api.autoinstallgen.generate_autoinstall(**autoinstall_args)
``` 

Still getting:
```
cobbler/actions/buildiso/standalone.py:160:70 - error: Argument of type "Profile | System" cannot be assigned to parameter "profile" of type "Profile | None" in function "generate_autoinstall" (reportGeneralTypeIssues)
cobbler/actions/buildiso/standalone.py:160:70 - error: Argument of type "Profile | System" cannot be assigned to parameter "system" of type "System | None" in function "generate_autoinstall" (reportGeneralTypeIssues)
```

TBH I'm fine either going with `# type: ignore` or preserving the `cast()`, but following Alex's reasoning, I wouldn't compromise readability/complexity of our code just because `pyright` not being able to properly determine the type here, so probably slighly in favor of `# type: ignore`, so I'm pushing it.